### PR TITLE
AMI 430 VISA driver: add retry to write and ask

### PR DESCRIPTION
**Pull request by jenshnielsen**
The communication with the AMI430 has been found to be somewhat flaky in some network setups. This seems to improve matters by retrying the communication

### CODE REVIEW

File: qcodes/instrument_drivers/american_magnetics/AMI430_visa.py [hunk 1]
```diff
@@ -0,0 +1,1110 @@
...
+    Driver for the American Magnetics Model 430 magnet power supply programmer.
+
+    This class controls a single magnet power supply. In order to use two or
+    three magnets simultaneously to set field vectors, first instantiate the
+    individual magnets using this class and then pass them as arguments to
+    either the AMI430_2D or AMI430_3D virtual instrument classes.
+
+    Args:
+        name: a name for the instrument
+        address: IP address of the power supply programmer
```
Comment (astafan8) [inline qcodes/instrument_drivers/american_magnetics/AMI430_visa.py:151]:
```suggestion
        address: visa address of the power supply programmer
```

File: qcodes/instrument_drivers/american_magnetics/AMI430_visa.py [hunk 2]
```diff
@@ -0,0 +1,1110 @@
...
+
+    def _connect(self) -> None:
+        """
+        Append the IPInstrument connect to flush the welcome message of the AMI
+        430 programmer
+        :return: None
+        """
+        super()._connect()
+        self.flush_connection()
+
```
Comment (astafan8) [inline qcodes/instrument_drivers/american_magnetics/AMI430_visa.py:425]: no need anymore?
```suggestion
```

File: qcodes/instrument_drivers/american_magnetics/AMI430_visa.py [hunk 3]
```diff
@@ -0,0 +1,1110 @@
...
+    _SHORT_UNITS = {"seconds": "s", "minutes": "min", "tesla": "T", "kilogauss": "kG"}
+    _DEFAULT_CURRENT_RAMP_LIMIT = 0.06  # [A/s]
+
+    def __init__(
+        self,
+        name: str,
+        address: str,
+        reset: bool = False,
+        terminator: str = "\r\n",
```
Comment (astafan8) [inline qcodes/instrument_drivers/american_magnetics/AMI430_visa.py:163]: noticed that the terminator used in the tests with the sim is actually ``\n``. could be relevant or not, dont know

Comment (jenshnielsen) [reply qcodes/instrument_drivers/american_magnetics/AMI430_visa.py:163]: I was wondering about that too. Looking at the log message in the traceback the read terminator is clearly \r\n

Comment (jenshnielsen) [reply qcodes/instrument_drivers/american_magnetics/AMI430_visa.py:163]: ValueError: ("could not convert string to float: '0.001666667,1.0\\r\\n'", 'setting magnet_x_field to 0.05')

**Review by astafan8 — APPROVED** [review]

**Comment by astafan8** [comment]
bors merge

**Comment by astafan8** [comment]
bors merge

**Pull request merged**